app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,62 @@
+class TripsController < ApplicationController
+  def index
+    @trips = Trip.all
+  end
+
+  def show
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      head :not_found
+    end
+  end
+
+  def create
+    if params[:passenger_id]
+      passenger = Passenger.find_by(id: params[:passenger_id])
+
+
+      driver = (Driver.all.select { |d| d.availability == true }).sample
+
+      @trip = passenger.trips.new(driver_id: driver.id, passenger_id: passenger.id, date: Date.today, rating: 0, cost: 0)
```

Comment:
All of the work done in this method could be moved to a model method. An instance method on `Passenger` would probably work well, something like `Passenger#request_trip`.